packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts

```diff
@@ -465,6 +465,9 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
             }
             const part = containerWidget.getPartFor(widget);
             if (part) {
+                part.node.addEventListener('mouseenter', () => {
```

Comment:
Could you explain the reasoning for setting this value on `mouseenter`? `mouseenter` doesn't generally correspond to other definitions of widget focus. What scenario does this account for?

Reply:
This is mainly to resolve the issue mentioned in below comment. As mentioned, the viewId now will not be updated as part of isVisible. So instead we would need a place to update viewContextKeys on view hover (toolbar items are shown on hover).

Reply:
I see what you mean, thanks for the explanation. Should this stay here and only apply to plugins, or could this also be a problem for 'native' view containers, so that it should be part of the code for the `ViewContainer` or `ViewContainerPart` classes? Are there any side effects to setting the viewID in this way - for example will the setting persist longer than intended?

Reply:
Since the isVisible clearly bound to PluginViewWidget (this was the place previously view context been updated) that's the reason i added specific to plugins. If needed i can look into making generic.

Reply:
Regarding side effects will do a further in-depth round of testing. Till now i haven't found any.